docs/src/main/asciidoc/security-oidc-bearer-token-authentication-tutorial.adoc

```diff
@@ -231,7 +231,10 @@ For more information, see the Keycloak documentation about link:https://www.keyc
 ifndef::no-quarkus-keycloak-admin-client[]
 [NOTE]
 ====
-If you want to use the Keycloak Admin Client to configure your server from your application, you need to include either the `quarkus-keycloak-admin-rest-client` or the `quarkus-keycloak-admin-resteasy-client` (if the application uses `quarkus-rest-client`) extension.
+To use the Keycloak Admin Client for configuring the Keycloak server from your application, you must include either the `quarkus-keycloak-admin-rest-client` or the `quarkus-keycloak-admin-resteasy-client` extension.
+
+Choose `quarkus-keycloak-admin-rest-client` if you are using `quarkus-rest-client`, and `quarkus-keycloak-admin-resteasy-client` if you are using `quarkus-resteasy-client`.
```

Comment:
It is true statement for subset of scenarios. You don't need to use `quarkus-rest-client` in order to use `quarkus-keycloak-admin-rest-client`, because the admin client brings the `quarkus-rest-client`. So yes, use _if you already have a REST client, you must use  `quarkus-keycloak-admin-rest-client`_, but also _use  `quarkus-keycloak-admin-rest-client` if you use Quarkus REST_ and _use `quarkus-keycloak-admin-resteasy-client` if you already use Quarkus RESTEasy_. To clarify, you can use the Admin client in following situations:

- with REST server
- with REST client
- with REST server and client

Reply:
I think we should go with something like (obviously to be improved, I'm not a writer :)):

If you are using Quarkus REST (`quarkus-rest` and/or `quarkus-rest-client`), use `quarkus-keycloak-admin-rest-client`. If you are using RESTEasy Classic (...), use ....

If you are not using any of the REST layers explicitly, it is recommended to use `quarkus-keycloak-admin-rest-client`.

Reply:
Thank you both, @michalvavrik and @gsmet. I've updated the text accordingly.